### ISSUE TITLE
Filters update fails

### ISSUE THREAD

**Issue by innit86** [AdguardTeam/AdguardForAndroid#5321]
### Please answer the following questions for yourself before submitting an issue

- [X] Filters were updated before reproducing an issue
- [X] I checked the [knowledge base](https://adguard.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.4 Nightly 46

### Environment

- OS version: Android 14
- Device: Pixel 7 Pro

### HTTPS filtering

- [X] yes, I do

### Root access

- [X] yes, I have it

### Integration with AdGuard VPN

- [ ] yes, I do

### Routing mode

Automatic proxy

### Which DNS server do you use?

Custom DNS

### DNS protocol

DNS-over-HTTPS

### Issue Details

Steps to reproduce:
1. Tap update filters.
2. Some filters don't update.
3.

### Expected Behavior

Filters should update.

### Actual Behavior

Some filters fail to update.   

### Screenshots

![Screenshot_20240515-172302~2.png](https://github.com/AdguardTeam/AdguardForAndroid/assets/58744017/505c7b9d-48b6-4aab-8315-5b2e5da76f06)

### Additional Information

Show on update page which filters and userscripts fail to update so that they could be disabled or removed.

**Comment by tengizmax** [comment]
Thank you for reporting this issue. 

We're looking forward to tell you some about UX accessibility of our app. To see additional info about failed updates you can tap on it (you marked it by red arrow on your screenshot).

To help us understand the problem better, could you please answer a few questions?
1) Can you please specify which filter is not updating?
2) Have you tried switching to a different Wi-Fi network or using mobile data and update filters again ?

Your answers will help us investigate this issue further.

**Comment by innit86** [comment]
> Thank you for reporting this issue. 
> 
> We're looking forward to tell you some about UX accessibility of our app. To see additional info about failed updates you can tap on it (you marked it by red arrow on your screenshot).
> 
> To help us understand the problem better, could you please answer a few questions?
> 1) Can you please specify which filter is not updating?
> 2) Have you tried switching to a different Wi-Fi network or using mobile data and update filters again ?
> 
> Your answers will help us investigate this issue further. 

I don't know which filters are not updating because tapping on "content blocker filters", "dns filters" & "userscripts" does not reveal it.

This is why I asked you to implement a feature that will allow user to see clearly which filters don't work and fail to update.

For example:

1. Tap update filters.
2. Some filters update and some don't.
3. Tap on the category in which some filters failed to update.
4. User is presented with a list of all filters that properly updated and those that didn't update.
5. User now know the culprit and can disable or remove the non-working filters or userscripts.

**Comment by Gestrid** [comment]
This "bug" seems to have made it into the stable release as well.

I was experiencing the same "userscripts failed to update" error, so I went into my userscripts settings and disabled them one at a time. I found the cause of my specific error. One of the userscripts I'd installed now leads to a 404 page. For some reason, prior to updating to v4.4, this did not throw an error. Now, it does.

The 404 is not caused by AdGuard. The user who created the userscript was apparently blocked on GitLab, and their content was removed from the site.

I can't confirm this for myself since all my filters (including my custom ones) are working properly, but perhaps one of OP's filters now leads to a 404 page when trying to update?
